ouroboros-consensus/src/Ouroboros/Consensus/ChainSyncClient.hs

```diff
@@ -424,8 +427,12 @@ chainSyncClient tracer cfg btime (ClockSkew maxSkew)
 
           -- TODO: Chain sync Client: Reuse anachronistic ledger view? #581
           case anachronisticProtocolLedgerView cfg curLedger (pointSlot hdrPoint) of
-            Nothing   -> retry
-            Just view -> case view `SB.at` hdrSlot of
+            -- unexpected alternative; see comment before this case expression
```

Comment:
> we should disconnect in this case

Is there a document that specifies that sort of thing?

In particular, what's the consequence of disconnecting? EG If we never reconnect, then consensus may never be reached. And note that this exception is being raised just because a new node is joining a network that's been around awhile -- that's a realistic case.

On the current tip of this branch, this particular exception causes the ChainSync client to terminate and then be restarted at the beginning of the next slot -- eventually the peer will "catch up" (since the ChainSync in the other direction between these two nodes is not stuck) and after one or more iterations of that, this exception eventually won't be raised again. Thus the nodes can still reach consensus.

Reply:
The new Issue 998 will resolve this